telcell/models/simplemodel.py

```diff
@@ -125,40 +132,147 @@ def sort_key(element):
+    location_counts = Counter(
+        location_key(m) for m in history_outside_interval)
+    min_rarity, rarest_pair = min(
+        ((location_counts.get(location_key(pair.measurement_b), 0), pair)
+         for pair in pairs_in_interval), key=sort_key)
     return rarest_pair
 
 
+def calculate_distance_for_pair(pair: MeasurementPair) -> float:
+    """
+    Calculate the distance between two measurements of a given pair.
+    @param pair: the pair to calculate the distance for.
+    @return: the calculated distance
+    """
+    latlon_a = pair.measurement_a.latlon
+    latlon_b = pair.measurement_b.latlon
+    return calculate_distance_lat_lon(latlon_a, latlon_b)
+
+
+def calculate_distance_lat_lon(latlon_a: Tuple[float, float],
+                               latlon_b: Tuple[float, float]) -> float:
+    """
+    Calculate the distance between a set of lat-lon coordinates.
+    @param latlon_a: the latitude and longitude of the first object
+    @param latlon_b: the latitute and longitude of the second object
+    @return: the calculated distance
+    """
+    geod = pyproj.Geod(ellps='WGS84')
+    lat_a, lon_a = latlon_a
+    lat_b, lon_b = latlon_b
+    _, _, distance = geod.inv(lon_a, lat_a, lon_b, lat_b)
+    return distance
+
+
+def select_colocated_pairs(tracks: List[Track],
+                           min_delay: datetime.timedelta = datetime.timedelta(
+                               seconds=0),
+                           max_delay: datetime.timedelta = datetime.timedelta(
+                               seconds=120)) -> List[MeasurementPair]:
+    """
+    For a list of tracks, find pairs of measurements that are colocated, i.e.
+    that do not share the same track name, but do share the owner. Also filter
+    the pairs based on a minimum and maximum time delay.
+    @param tracks: the tracks to find pairs of.
+    @param min_delay: the minimum amount of delay that is allowed.
+    @param max_delay: the maximum amount of delay that is allowed.
+    @return: A filtered list with all colocated paired measurements.
+    """
+    final_pairs = []
+    for track_a, track_b in combinations(tracks, 2):
+        if track_a.owner == track_b.owner and track_a.name != track_b.name:
+            pairs = pair_measurements_based_on_time(track_a, track_b)
+            pairs = filter_delay(pairs, min_delay, max_delay)
+            final_pairs.extend(pairs)
+    return final_pairs
+
+
+def generate_dislocated_pairs(measurement: Measurement, track: Track) \
+        -> List[MeasurementPair]:
+    """
+    Created paired measurements that are dislocated. We do so by linking one
+    specific measurement to every measurement of a given track.
+    @param measurement: the measurement that will be linked to other
+     measurements
+    @track: the measurements of this track will be linked to the given
+     measurement
+    @return: A list with dislocated paired measurements.
+    """
+    pairs = []
```

Comment:
Check of de measurement daadwerkelijk niet in de track zit (in measurement.extra zit nog de 'track' en 'sensor')